src/python/turicreate/toolkits/sound_classifier/sound_classifier.py

```diff
@@ -105,14 +106,47 @@ def reset(self):
         # instance itself.
         pass
 
+class _NumPyDataIterator(_DataIterator):
+    def __init__(self, data, label=None, batch_size=1, shuffle=False):
+
+        # Always pass a tuple, so that the impl's built-in iterator returns a
+        # tuple.
+        self.data_slices = (data, label) if label is not None else (data,)
+        self.batch_size = batch_size
+        self.batch_idx = 0
+        self.num_batches = int(_ceil(len(data) / float(self.batch_size)))
+
+    def __iter__(self):
+        if self.batch_idx < self.num_batches:
+            data = self.data_slices[0][self.batch_size*self.batch_idx:self.batch_size*(self.batch_idx+1)]
+            label = None
+            if len(self.data_slices)>1:
+                label = self.data_slices[1][self.batch_size*self.batch_idx:self.batch_size*(self.batch_idx+1)]
+            self.batch_idx += 1
+            yield data, label if label is not None else data
```

Comment:
Might need to return a tuple in the else case, `data,` instead of `data`

Reply:
It automatically converts the return in the else case as a tuple. Which is why predict needs to extract `data[0]`